projects/storefrontlib/shared/services/import-export/export-csv.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { FileOptions } from '../../models/file';
+import { ExportService } from './export.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExportCsvService {
+  constructor(protected exportService: ExportService) {}
+  /**
+   * Converts array of objects into CSV data structure.
+   *
+   * @param objectsArray Array of objects which should be converted to CSV.
+   * @param separator for csv data
+   * @returns Processed string ready to be saved into file
+   */
+  dataToCsv(objectsArray: string[][], separator: string): string {
+    const array =
+      typeof objectsArray != 'object' ? JSON.parse(objectsArray) : objectsArray;
```

Comment:
Smell: Why do we need to JSON.parse stuff that is already of type `string[][]`? It seems like we're abusing some typing? Moreover the name `objectsArray` suggests it's an array of objects.